src/GenericContainer/include/BipedalLocomotionControllers/GenericContainer.h

```diff
@@ -0,0 +1,289 @@
+/**
+ * @file GenericContainer.h
+ * @authors Stefano Dafarra
+ * @copyright 2019 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#ifndef BIPEDAL_LOCOMOTION_CONTROLLERS_GENERICCONTAINER_H
+#define BIPEDAL_LOCOMOTION_CONTROLLERS_GENERICCONTAINER_H
+
+#include <iDynTree/Core/Span.h>
+
+#include <BipedalLocomotionControllers/TemplateHelpers.h>
+
+#include <functional>
+#include <cassert>
+#include <iostream>
+
+namespace BipedalLocomotionControllers {
+    template <typename T>
+    class GenericContainer;
+}
```

Comment:
Probably there is a namespace missing. 

```cpp
namespace BipedalLocomotionControllers 
{
namespace GenericContainer 
{
    template <typename T>
    class GenericContainer;
}
}
```

Too verbose right?

Reply:
in C++17 we can also use
```cpp
namespace BipedalLocomotionControllers::GenericContainer
{
    template <typename T>
    class GenericContainer;
}
```

Reply:
To be completely honest, I am not a huge fan of duplicated names. In general, I think we may be a little verbose. Without a ``using`` we would have
```c++
BipedalLocomotionController::GenericContainer::GenericContainer c = BipedalLocomotionController::GenericContainer::makeGenericContainer(v);
```
but on the other hand, I like if we have symmetry among components and included file (i.e. the namespace follows the inclusion path). We could use another name, maybe dropping "Generic". What would you prefer?

Reply:
Something like?
```cpp
namespace BipedalLocomotionControllers::GenericContainer
{
    template <typename T>
    class Container;
}
```

Reply:
Nitpick: Container is quite a generic name, and  in C++ it has a precise meaning, see https://en.cppreference.com/w/cpp/container .  This class seems to be something that is meant just to wrap vectors that can (or cannot) be resized. For this reason I think  that a name that recall vector or array would be more clear (something like VectorWrapper or GenericVector). However, you already put a lot of effort on this, so if it permits you to go faster feel free to proceed with "GenericContainer".

Reply:
The hope at the beginning was that it could have been used also with ``deque``, but I did not realize that it does not have a ``data`` method, hence it cannot be used with ``span``. On hindsight, this makes sense since data in ``deque`` is not contiguous. 
Anyways, I was already planning to change the name. Given the three comments above, I could name it ``Vector`` in the namespace ``GenericContainer``.

Reply:
I have renamed it as
```cpp
namespace BipedalLocomotionControllers::GenericContainer
{
    template <typename T>
    class Vector;
}
```